examples/Example17/src/EventAction.cc

```diff
@@ -65,9 +77,33 @@ void EventAction::BeginOfEventAction(const G4Event *)
 
 void EventAction::EndOfEventAction(const G4Event *aEvent)
 {
+  auto eventId         = G4EventManager::GetEventManager()->GetConstCurrentEvent()->GetEventID();
 
   fTimer.Stop();
 
+  #if defined BENCHMARK
+    //Get the Run object associated to this thread and stop the timer for this event
+    Run* currentRun = static_cast< Run* > ( G4RunManager::GetRunManager()->GetNonConstCurrentRun() );
+    auto aBenchmarkManager = currentRun->getBenchmarkManager();
+    aBenchmarkManager->timerStop(Run::timers::EVENT);
+
+    //Accumulate the timings
+    double eventTime = aBenchmarkManager->getDurationSeconds(Run::timers::EVENT);
+    double nonEMTime = aBenchmarkManager->getDurationSeconds(Run::timers::NONEM_EVT);
+    double ecalTime = eventTime - nonEMTime;
+    
+    aBenchmarkManager->addDurationSeconds(Run::timers::EVENT_SUM, eventTime);
+    aBenchmarkManager->addDurationSeconds(Run::timers::EVENT_SQ, pow(eventTime, 2));
```

Comment:
Avoid calling the power function for squared values, here it does not matter much, but in performance-critical code it may. It is not granted that the compilers will do the optimization for you [see here](https://stackoverflow.com/questions/2940367/what-is-more-efficient-using-pow-to-square-or-just-multiply-it-with-itself)